### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by BlueShadow232**
Merge Intake into main

### CODE REVIEW

File: src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+  init {
+    intakeTalon.configFactoryDefault()
+    Logger.addSource("INTAKE", "Intake State") { intakeState.toString() }
+    Logger.addSource("INTAKE", "Intake Motor Power") { intakeTalon.motorOutputPercent }
+    Logger.addSource("INTAKE", "Intake Motor Stator Current") { intakeTalon.statorCurrent }
+    Logger.addSource("INTAKE", "Intake Motor Supply Current") { intakeTalon.supplyCurrent }
+    Logger.addSource("INTAKE", "Intake Motor Voltage") { intakeTalon.motorOutputVoltage }
+    Logger.addSource("INTAKE", "Arm State") { armState.toString() }
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt]: we want to be consistent with the tabs so make it lowercase

File: src/main/kotlin/com/team4099/robot2022/commands/intake/intakeCommand.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.team4099.robot2022.commands.intake
+
+import com.team4099.lib.logging.Logger
+import com.team4099.robot2021.subsystems.com.team4099.robot2022.subsystems.Intake
+import com.team4099.robot2022.config.Constants
+import edu.wpi.first.wpilibj2.command.CommandBase
+
+class IntakeCommand(
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/commands/intake/intakeCommand.kt]: fix file name case

File: src/main/kotlin/com/team4099/robot2022/commands/intake/intakeIdleCommand.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.team4099.robot2022.commands.intake
+
+import com.team4099.lib.logging.Logger
+import com.team4099.robot2021.subsystems.com.team4099.robot2022.subsystems.Intake
+import com.team4099.robot2022.config.Constants
+import edu.wpi.first.wpilibj2.command.CommandBase
+
+class IntakeIdleCommand : CommandBase() {
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/commands/intake/intakeIdleCommand.kt]: fix casing for file name

File: src/main/kotlin/com/team4099/robot2022/Robot.kt [hunk 4]
```diff
@@ -2,15 +2,22 @@ package com.team4099.robot2022
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/Robot.kt]: i cant comment but idk what the .DS_Store files are but we shouldnt need them

File: src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    intakeTalon.configOpenloopRamp(Constants.Intake.RAMP_TIME)
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt:47]: on the 2021 bot we were having problems with intake drawing a lot of power so maybe add a limit for the amount of current

File: src/main/kotlin/com/team4099/robot2022/config/Constants.kt [hunk 6]
```diff
@@ -29,6 +30,26 @@ object Constants {
             0 to RobotName.COMPETITION, 1 to RobotName.PRACTICE, 2 to RobotName.MULE)
   }
 
+  object Intake {
+    const val MOTORID = 1 // Need to change this later
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/config/Constants.kt]: what is the purpose of MOTORID as doesnt intake_motor do the same thing

**Review by eckshn — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt (hunk 1 above)
Comment (eckshn) [reply src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt]: actually make a constant for the tab name

Comment (BlueShadow232) [reply src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt]: I made the INTAKE lowercase, do you also want the Intake State, Power, etc to be lower case too?

File: src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import com.team4099.lib.logging.Logger
+import com.team4099.robot2022.config.Constants
+import edu.wpi.first.wpilibj.DoubleSolenoid
+import edu.wpi.first.wpilibj.PneumaticsModuleType
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+object Intake : SubsystemBase() {
+
+  // motor params
+  private val steeringConfiguration: TalonFXConfiguration = TalonFXConfiguration()
+  private val driveConfiguration: TalonFXConfiguration = TalonFXConfiguration()
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt:15]: change names and you only need one

File: src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    Logger.addSource(Constants.Intake.TAB, "Intake State") { intakeState.toString() }
+    Logger.addSource(Constants.Intake.TAB, "Intake Motor Power") { intakeTalon.motorOutputPercent }
+    Logger.addSource(Constants.Intake.TAB, "Intake Motor Stator Current") { intakeTalon.statorCurrent }
+    Logger.addSource(Constants.Intake.TAB, "Intake Motor Supply Current") { intakeTalon.supplyCurrent }
+    Logger.addSource(Constants.Intake.TAB, "Intake Motor Voltage") { intakeTalon.motorOutputVoltage }
+    Logger.addSource(Constants.Intake.TAB, "Arm State") { armState.toString() }
+
+    steeringConfiguration.supplyCurrLimit.currentLimit =
+      Constants.Intake.STEERING_SUPPLY_CURRENT_LIMIT
+    intakeTalon.configOpenloopRamp(Constants.Intake.RAMP_TIME)
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/subsystems/Intake.kt:47]: you need to call the method thats called like intakeTalon.configAll(intakeConfiguration)

File: gradlew.bat [hunk 9]
```diff
@@ -1,89 +1,89 @@
...
-rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
-rem the _cmd.exe /c_ return code!
-if  not "" == "%GRADLE_EXIT_CONSOLE%" exit 1
-exit /b 1
-
-:mainEnd
-if "%OS%"=="Windows_NT" endlocal
-
-:omega
+@rem
```
Comment (eckshn) [inline gradlew.bat:1]: in command prompt once you have made all your changes do
./gradlew :spotlessApply

Comment (eckshn) [reply gradlew.bat:1]: this was just a random file i chose to comment on btw

File: src/main/kotlin/com/team4099/robot2022/config/Constants.kt [hunk 10]
```diff
@@ -29,6 +30,28 @@ object Constants {
             0 to RobotName.COMPETITION, 1 to RobotName.PRACTICE, 2 to RobotName.MULE)
   }
 
+  object Intake {
+    const val INTAKE_MOTOR = 1 // Change
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/config/Constants.kt:34]: update this to 31

**Pull request closed without merging**